lib/daru/dataframe.rb

```diff
@@ -1824,14 +1824,15 @@ def merge other_df # rubocop:disable Metrics/AbcSize
 
       new_fields = (@vectors.to_a + other_df.vectors.to_a)
       new_fields = ArrayHelper.recode_repeated(new_fields)
-
-      DataFrame.new({}, order: new_fields).tap do |df_new|
+      temp = DataFrame.new({}, order: new_fields).tap do |df_new|
         (0...nrows).each do |i|
           df_new.add_row row[i].to_a + other_df.row[i].to_a
         end
 
         df_new.update
       end
+      temp.index= @index if @index == other_df.index
+      temp
```

Comment:
Oh, sorry for not noticing it earlier. You can reuse that `tap` block that was here before (it is exactly the idiom for avoiding "temp" local variables:
```ruby
DataFrame.new({}, order: new_fields).tap do |df_new|
  (0...nrows).each do |i|
    df_new.add_row row[i].to_a + other_df.row[i].to_a
  end

  df_nex.index = @index if @index == other_df.index

  df_new.update
end
```

Reply:
oh got it! :+1: